Microsoft.Toolkit.HighPerformance/Extensions/HashCodeExtensions.cs

```diff
@@ -23,12 +23,7 @@ public static class HashCodeExtensions
         /// <param name="span">The input <see cref="ReadOnlySpan{T}"/> instance.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void Add<T>(ref this HashCode hashCode, ReadOnlySpan<T> span)
-#if SPAN_RUNTIME_SUPPORT
```

Comment:
I've removed that restriction because I've now added polyfills for the `RuntimeHelpers.IsReferenceOrContainsReferences<T>()` API on .NET Standard 2.0 and below (I needed this for other code in the `Memory2D<T>` and `Span2D<T>` types, so I used it here too while I was at it), I use that here:

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/2745962d560c4cf96b55074bc8babd0da7c6b0b3/Microsoft.Toolkit.HighPerformance/Helpers/HashCode%7BT%7D.cs#L62-L65

Now that older targets can test this too at runtime, we no longer have to add that extra type constraint, which was basically just a precaution due to the lack of this API at runtime. You can see how we switch the polyfill on here:

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/2745962d560c4cf96b55074bc8babd0da7c6b0b3/Microsoft.Toolkit.HighPerformance/Helpers/HashCode%7BT%7D.cs#L12-L16

So now on all targets the API will just automatically pick the right implementation for each input type `T` 😄